client/client.go

```diff
@@ -256,6 +246,116 @@ func (r *NotaryRepository) Initialize(rootKeyIDs []string, serverManagedRoles ..
 	return r.saveMetadata(serverManagesSnapshot)
 }
 
+// certsOfKeyIDs either confirms that the certs and keys (represented by Key IDs) forms valid key pairs.
+// Or throw error when they missmatch.
+// Or generate certificates from the keys if no certificate is provided
+func (r *NotaryRepository) certsOfKeyIDs(keyIDs []string, certs []data.PublicKey) ([]data.PublicKey, error) {
```

Comment:
David, just to confirm what you meant. At the moment 'certsOfKeyIDs` does not generate certs if `--rootcert` flag is provided. It will only generate certs in the `notary init --rootkey key` scenario. 

it will return an error if 1) certs does not match privKeys 2) inconsistent number of certs and keys provided and 3) non-matching key-cert pairs. 

Do you mean to simplify this method by moving cert generation to somewhere else and only return an `error` if certs and IDs does not match?